client/service/src/client/client.rs

```diff
@@ -1037,9 +1061,28 @@ where
 			);
```

Comment:
This code should only be called when `ImportNotificationAction::Both` or `ImportNotificationAction::RecentBlock`